tests/zkevm/test_worst_compute.py

```diff
@@ -955,3 +956,40 @@ def test_empty_block(
         post={},
         blocks=[Block(txs=[])],
     )
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "empty_calldata",
+    [False, True],
+)
+def test_worst_calldataload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    empty_calldata: bool,
+):
+    """Test running a block with as many CALLDATALOAD as possible."""
+    env = Environment()
+
+    code_prefix = Op.JUMPDEST
+    code_suffix = Op.PUSH0 + Op.JUMP
+    code_body_len = MAX_CODE_SIZE - len(code_prefix) - len(code_suffix)
+    code_loop_iter = Op.POP(Op.PUSH0 + Op.CALLDATALOAD)
+    code_body = code_loop_iter * (code_body_len // len(code_loop_iter))
+    code = code_prefix + code_body + code_suffix
```

Comment:
```suggestion
    code = code_prefix + code_body + Op.JUMPDEST * (code_body_len % len(code_loop_iter)) + code_suffix
```

(just stumbled on this PR, could fill with JUMPDEST to pass the `MAX_CODE_SIZE` assertion)

Reply:
Ah, I think I prefer to change the `==` to `<=`, just to avoid extra unrelated gas.